CARDS-1420: Add Subject Type filters similar to the existing Subject filter

acrowthe opened the pull request:
- Adds an additional "Subject Type" filter for each subject type
<img width="909" height="525" alt="image" src="https://github.com/user-attachments/assets/171e6a9a-0c97-4bc2-bbf4-63daf8e79100" />
<img width="902" height="348" alt="image" src="https://github.com/user-attachments/assets/fcaac7cc-b636-4148-ad4a-68b94300d185" />

acrowthe committed:
CARDS-1420: Split the Subject filter into separate filters by subject type
- Splits the subject filter into one filter per subject type

sdumitriu reviewed (commented):
Regression or existing bug? Filters are not preserved when expanding the table.

acrowthe committed:
CARDS-1420: Split the Subject filter into separate filters by subject type
- Fix filters being lost when expanding table

acrowthe committed:
CARDS-1420: Add a Subject Type filter similar to the existing Subject filter
- Adds an additional "Subject Type" filter for each subject type

acrowthe committed:
CARDS-1420: Add a Subject Type filter similar to the existing Subject filter
- Reorder Subject and SubjectType filters to avoid duplicate Autocomplete Header warning
